include/rapidcheck/gen/Build.hpp

```diff
@@ -119,15 +119,51 @@ class BuildMapper<T, rc::detail::IndexSequence<Indexes...>, Lenses...> {
   std::tuple<Lenses...> m_lenses;
 };
 
+template<typename T, typename... Args>
+typename std::enable_if<std::is_constructible<T, Args...>::value, T>::type
+construct(Args &&... args) {
+  return T(std::forward<Args>(args)...);
+}
+
+template<typename T, typename... Args>
+typename std::enable_if<!std::is_constructible<T, Args...>::value, T>::type
+construct(Args &&... args) {
+  return T{std::forward<Args>(args)...};
+}
+
+template<typename T, typename... Args>
+typename std::enable_if<std::is_constructible<T, Args...>::value, std::unique_ptr<T>>::type
+makeUnique(Args &&... args) {
+  return std::unique_ptr<T>(new T(std::forward<Args>(args)...));
+}
+
+template<typename T, typename... Args>
+typename std::enable_if<!std::is_constructible<T, Args...>::value, std::unique_ptr<T>>::type
+makeUnique(Args &&... args) {
+  return std::unique_ptr<T>(new T{std::forward<Args>(args)...});
+}
+
+template<typename T, typename... Args>
+typename std::enable_if<std::is_constructible<T, Args...>::value, std::shared_ptr<T>>::type
+makeShared(Args &&... args) {
+  return std::shared_ptr<T>(new T(std::forward<Args>(args)...));
+}
+
+template<typename T, typename... Args>
+typename std::enable_if<!std::is_constructible<T, Args...>::value, std::shared_ptr<T>>::type
+makeShared(Args &&... args) {
+  return std::shared_ptr<T>(new T{std::forward<Args>(args)...});
```

Comment:
It will require checks for `std::is_copy_constructible` and `std::is_move_constructible`, since it's possible to have aggregates that are not copyable/movable. @emil-e, do you think it's worth adding extra overloads?

Reply:
I'm not sure I think it's worth it. This solution is not the absolutely most performant but it's the most compatible in combination with relative simplicity.

Reply:
I agree that it might be better to not strive for absolute generality for the sake of simplicity. And yeah, it will probably be completely insignificant in the larger picture.